Tests/SourceKitLSPTests/RenameTests.swift

```diff
@@ -759,4 +761,74 @@ final class RenameTests: XCTestCase {
     XCTAssertEqual(range, positions["1️⃣"]..<positions["2️⃣"])
     XCTAssertEqual(placeholder, "foo(a:b:)")
   }
+
+  func testGlobalRenameC() async throws {
+    try await assertMultiFileRename(
+      files: [
+        "Sources/MyLibrary/include/lib.h": """
+        void 1️⃣do2️⃣Stuff();
+        """,
+        "lib.c": """
+        #include "lib.h"
+
+        void 3️⃣doStuff() {
+          4️⃣doStuff();
+        }
+        """,
+      ],
+      language: .c,
+      newName: "doRecursiveStuff",
+      expected: [
+        "Sources/MyLibrary/include/lib.h": """
+        void doRecursiveStuff();
+        """,
+        "lib.c": """
+        #include "lib.h"
+
+        void doRecursiveStuff() {
+          doRecursiveStuff();
+        }
+        """,
+      ]
+    )
+  }
+
+  func testGlobalRenameObjC() async throws {
+    try await assertMultiFileRename(
+      files: [
+        "Sources/MyLibrary/include/lib.h": """
+        @interface Foo
+        - (int)1️⃣perform2️⃣Action:(int)action 3️⃣wi4️⃣th:(int)value;
+        @end
+        """,
+        "lib.m": """
+        #include "lib.h"
+
+        @implementation Foo
+        - (int)5️⃣performAction:(int)action 6️⃣with:(int)value {
+          return [self 7️⃣performAction:action 8️⃣with:value];
+        }
+        @end
+        """,
+      ],
+      language: .objective_c,
+      newName: "performNewAction:by:",
+      expected: [
+        "Sources/MyLibrary/include/lib.h": """
+        @interface Foo
+        - (int)performNewAction:(int)action by:(int)value;
+        @end
+        """,
+        "lib.m": """
+        #include "lib.h"
+
+        @implementation Foo
+        - (int)performNewAction:(int)action by:(int)value {
+          return [self performNewAction:action by:value];
+        }
+        @end
+        """,
+      ]
+    )
+  }
```

Comment:
Could we add a cross-language test? Or multiple even, eg. swift <-> objc, swift <-> c, and swift <-> c++?

Reply:
Ha, no. This PR is about adding global rename for clang languages, not doing cross-language rename yet. For that we need name translation, which I’ll add in a follow-up PR (and which is currently blocked by rdar://120066209).

Reply:
IIUC only ObjC is blocked by that right 😅? But cool, didn't realize we didn't have name translation yet, thanks 👍

Reply:
Yeah, but ObjC is really the interesting candidate for name translation. And it’s easiest to design it with all name translations in mind. There will be a follow-up PR soon-ish.